src/runtime/rpc/rpc_host_port.cpp

```diff
@@ -113,4 +142,43 @@ void host_port::assign_group(const char *name)
     _group_host_port->add_ref();
 }
 
+error_s host_port::resolve_addresses(std::vector<rpc_address> &addresses) const
+{
+    CHECK(addresses.empty(), "invalid addresses, not empty");
+    if (type() != HOST_TYPE_IPV4) {
+        return std::move(error_s::make(dsn::ERR_INVALID_STATE, "invalid host_port type"));
+    }
+
+    rpc_address rpc_addr;
+    if (rpc_addr.from_string_ipv4(this->to_string().c_str())) {
```

Comment:
Add some comments to explain why we can skip the following resolve logic if call `from_string_ipv4` succeed.